apps/api/v2/src/ee/calendars/services/gcal.service.ts

```diff
@@ -149,10 +144,30 @@ export class GoogleCalendarService implements OAuthCalendarApp {
     const primaryCal = cals.data.items?.find((cal) => cal.primary);
 
     if (primaryCal?.id) {
-      await this.selectedCalendarsRepository.createSelectedCalendar(
-        primaryCal.id,
-        credential.id,
+      const alreadyExistingSelectedCalendar = await this.selectedCalendarsRepository.getUserSelectedCalendar(
+        ownerId,
+        GOOGLE_CALENDAR_TYPE,
+        primaryCal.id
+      );
+
+      if (alreadyExistingSelectedCalendar) {
```

Comment:
Question 2: why if we have found the calendar `if (alreadyExistingSelectedCalendar) {` we still need to call `await this.calendarsService.createAndLinkCalendarEntry` ?

Reply:
Why do we need to upsert the credentials? Why don't we have instead of upsert a create method and call it only if NOT alreadyExistingSelectedCalendar ?